hikari/internal/enums.pyi

```diff
@@ -33,11 +33,19 @@
 __all__ = ["Enum", "Flag"]
 
 import enum as __enum
+from typing import Any as __Any
 from typing import Iterator as __Iterator
 from typing import Sequence as __Sequence
 from typing import TypeVar as __TypeVar
 from typing import Union as __Union
 
+class deprecated:
+    value: __Any
+    removal_version: str
+
+    def __init__(self, value: __Any, /, *, removal_version: str): ...
```

Comment:
This doesn't seem to work as expected if you try to access the raw value using `ENUM_MEMBER.value`. At runtime this'll error since the enum member doesn't have a `value` attribute and at type checking the `value` attribute will be typed as `deprecated` (instead of the raw type).

Did you consider typing this like the following:

```py
def deprecated(value: _T, /, *, removal_version: str) -> _T:  ...
```

Where it'd be relying on the fact that type-checking logic would take the returned `_T` and wrap it in a specialisation of the enum's type for the class attribute while using _T for `EnumMember.value`.